front/pages/poke/[wId]/data_sources/[dsId]/index.tsx

```diff
@@ -106,6 +114,19 @@ export const getServerSideProps = withSuperUserAuthRequirements<{
     );
     if (connectorRes.isOk()) {
       connector = connectorRes.value;
+      const temporalClient = await getTemporalConnectorsNamespaceConnection();
+
+      const res = temporalClient.workflow.list({
```

Comment:
I fear this will slow down the page loading substantially (and it's already slow af).

Can we use a hook here?

Reply:
connectorId is indexed.
Let's see how it goes in prod and I'll gladly refactor to a hook if it's slow.